src/NServiceBus.AcceptanceTests/ConfigureEndpointMsmqTransport.cs

```diff
@@ -19,11 +20,24 @@ public class ConfigureScenariosForMsmqTransport : IConfigureSupportedScenariosFo
 
 public class ConfigureEndpointMsmqTransport : IConfigureEndpointTestExecution
 {
-    public Task Configure(string endpointName, EndpointConfiguration configuration, RunSettings settings)
+    public Task Configure(string endpointName, EndpointConfiguration configuration, RunSettings settings, PublisherMetadata publisherMetadata)
     {
         queueBindings = configuration.GetSettings().Get<QueueBindings>();
         var connectionString = settings.Get<string>("Transport.ConnectionString");
-        configuration.UseTransport<MsmqTransport>().ConnectionString(connectionString);
+        var transportConfig = configuration.UseTransport<MsmqTransport>();
+
+        transportConfig.ConnectionString(connectionString);
+
+        foreach (var publisher in publisherMetadata.Publishers)
+        {
+            foreach (var eventType in publisher.Events)
+            {
+                var publisherName = Conventions.EndpointNamingConvention(publisher.PublisherType);
+
+                transportConfig.Routing().RegisterPublisher(eventType, publisherName);
```

Comment:
not saying we really need to optimize this, but it would be better if we access `Routing()` only once.